x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/DelegatingTokenSource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.parser;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenFactory;
+import org.antlr.v4.runtime.TokenSource;
+
+/**
+ * Utility class for filtering/processing TokenSource.
+ */
+abstract class DelegatingTokenSource implements TokenSource {
```

Comment:
Extracted this base class since I had another TokenSource implementation - I've removed it but kept it for future reuse (or maybe ANTLR will offer it).